src/traits.jl

```diff
@@ -1,3 +1,70 @@
+"""
+    HasProperties(img) -> HasProperties{::Bool}
+
+Returns the trait `HasProperties`, indicating whether `x` has `properties`
+method.
+"""
+struct HasProperties{T} end
+
+HasProperties(img::T) where T = HasProperties(T)
+
+HasProperties(::Type{T}) where T = HasProperties{false}()
+
+"""
+    HasDimNames(img) -> HasDimNames{::Bool}
+
+Returns the trait `HasDimNames`, indicating whether `x` has named dimensions.
+Types returning `HasDimNames{true}()` should also have a `names` method that
+returns a tuple of symbols for each dimension.
+"""
+struct HasDimNames{T} end
+
+HasDimNames(img::T) where T = HasDimNames(T)
+
+HasDimNames(::Type{T}) where T = HasDimNames{false}()
+
+"""
+    namedaxes(img) -> NamedTuple{names}(axes)
+
+Returns a `NamedTuple` where the names are the dimension names and each indice
+is the corresponding dimensions's axis. If `HasDimNames` is not defined for `x`
+default names are returned. `x` should have an `axes` method.
+
+```jldoctest
+julia> using ImagesAPI
+
+julia> img = reshape(1:24, 2,3,4)
+
+julia> namedaxes(img)
+```
```

Comment:
For `jldoctest` you need to make sure the results are right:

```diff
-julia> using ImagesAPI
+julia> using ImageCore

-julia> img = reshape(1:24, 2,3,4)
+julia> img = reshape(1:24, 2,3,4);

julia> namedaxes(img)
+(:row, :col, :page))
```

Otherwise, there would be a warning when building the documentation with Documenter.